conversions/binary_to_decimal.c

```diff
@@ -1,24 +1,103 @@
+/** 
+ * @file
+ * @brief Converts a [binary to a decimal](https://byjus.com/binary-to-decimal-formula/) number implementation.
+ * @details
+ * A binary number is an input, it is checked to be binary
+ * then, the number is converted to a decimal number.
+ * @author [Kyler Smith](https://github.com/KylerSmith)
+ * @author [lazy-dude](https://github.com/lazy-dude)
+ */
+
+#include <assert.h>     /// for assert
+#include <stdbool.h>   /// for bool
+#include <stdint.h>   /// for uintmax_t
+#include <stdio.h>   /// for IO operations
+
 /**
- * Modified 07/12/2017, Kyler Smith
- *
+ * @brief checks whether the number is binary
+ * @param num the number to be checked if it has binary representation
+ * @returns true if the number IS binary
+ * @returns false if the number is NOT binary
  */
+bool is_binary(uintmax_t num)
+{
+    unsigned remainder = 0;
 
-#include <stdio.h>
+    while (num > 0) {
+        remainder = num % 10;
+        if (remainder == 0 || remainder == 1) {
+            num /= 10;
+            continue;
+        } else
+            return false;
+    }
+    return true;
+}
 
-int main()
+/**
+ * @brief finds the length of an `uintmax_t` number
+ * @param num whose length to be computed
+ * @returns the length of the number
+ */
+uint16_t num_len(uintmax_t num)
 {
-    int remainder, number = 0, decimal_number = 0, temp = 1;
-    printf("\n Enter any binary number= ");
-    scanf("%d", &number);
-
-    // Iterate over the number until the end.
-    while (number > 0)
-    {
-        remainder = number % 10;
-        number = number / 10;
-        decimal_number += remainder * temp;
-        temp = temp * 2;  // used as power of 2
+    uint16_t len = 0;
+    for (len = 0; num > 0; len++) {
+        num /= 10;
     }
+    return len;
+}
+
+/**
+ * @brief The `binary_decimal` function does the actual job of conversion
+ * @param the number binary to be converted
```

Comment:
A minor error here: we must provide the parameter's name then the description of the parameter.
```suggestion
 * @param number the number binary to be converted
```